### ISSUE TITLE
EC2 Error Code: RequestLimitExceeded

### ISSUE THREAD

**Issue by Puneeth-n** [terraform-providers/terraform-provider-aws#2942]
Hi,

Recently with `terraform 0.10.8` and `terraform-provider-aws 1.6` we started getting RequestLimitExceeded errors on our lambdas. We increased max_retries to 50 and still notice this issue. It occurs when we deploy lambdas. Specifically [here](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_lambda_function.go#L644)

```
Error modifying Lambda Function Configuration lambda-booking-generate_invoice: InvalidParameterValueException: Your request has been throttled by EC2, please make sure you have enough API rate limit. EC2 Error Code: RequestLimitExceeded. EC2 Error Message: Request limit exceeded.
```
Any idea how we can fix this issue?

**Comment by Puneeth-n** [comment]
cc: @radeksimko @apparentlymart

**Comment by bflad** [comment]
As a less than ideal workaround, you can reduce the parallelism that Terraform is running at via the `-parallelism=#` flag to `terraform apply`: https://www.terraform.io/docs/commands/apply.html#parallelism-n

It is 10 by default.

**Comment by bflad** [comment]
A potential code fix could be to put retry logic around this call catching `isAwsErr(err, lambda. ErrCodeInvalidParameterValueException, "Your request has been throttled by EC2")`

**Comment by Puneeth-n** [comment]
@bflad yes. You are right. Ideally, also the `max_tries` at the provider level should also solve this issue. However, the problem is, in the aws-sdk, `InvalidParameterValueException` is not a retryable exception [see](https://github.com/aws/aws-sdk-go/blob/07a12acf062c74aacf99ed8a6a4ee1bc52ab1627/aws/request/retryer.go#L35) thus, it fails. 

I am working on a PR to add the retry logic and some acceptance tests were failing. Will figure out the issue and create a PR here.

Furthermore, I created an issue in https://github.com/aws/aws-sdk-go/issues/1727

**Comment by bflad** [comment]
This throttling retries fix has landed in master and will be released as part of v1.7.1. Thanks for the PR!

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.1](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#171-january-19-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
